### DISCUSSION TITLE
Requesting additional OAuth provider scopes after login with Supabase

### DISCUSSION THREAD

**Discussion by lyqht** [supabase/supabase#7352] [Questions]
Hello there, I'm building a Next.js project, [Unfork](https://github.com/lyqht/unfork), which allows people to easily identify the forks they have and delete them to declutter their GitHub profile. I have a user who has the raised [the concern that I'm requesting the `delete_repo` permission at login itself](https://twitter.com/paradite_/status/1538916962654654466/photo/1), which applies to all repositories and not just the specific repositories that they would later allow to be deleted.

This is because of the current code

```
export const LoginButton = () => (
  <Button
    onClick={() =>
      supabase.auth.signIn(
        { provider: `github` },
        { scopes: `delete_repo, repo` },
      )
    }
  >
    Sign in with GitHub
  </Button>
);
```

At the moment from reading GitHub's documentation, it is not possible to request delete_repo scope only for specific repos, however I would like to at least only request the delete_repo scope from the user only after they have viewed and chosen at least 1 repository to delete. Is this possible with Supabase?

**Comment by zdanl** [comment]
Please update this ticket.

**Comment by ovy9086** [comment]
any updates on this ? were you able to achieve this or workaround this limitation ?

**Comment by danilhendrasr** [comment]
Any updates on this?
@lyqht have you found a solution?
